src/core/mavlink_commands.h

```diff
@@ -122,19 +123,20 @@ class MavlinkCommandSender {
         mavlink_message_t mavlink_message{};
         CommandResultCallback callback{};
         dl_time_t time_started{};
+        void* timeout_cookie = nullptr;
 
         explicit Work(double new_timeout_s) : timeout_s(new_timeout_s) {}
     };
 
     void receive_command_ack(mavlink_message_t message);
-    void receive_timeout();
+    void receive_timeout(const uint16_t command);
 
     void call_callback(const CommandResultCallback& callback, Result result, float progress);
 
     SystemImpl& _parent;
     LockedQueue<Work> _work_queue{};
-
-    void* _timeout_cookie = nullptr;
+    std::unordered_map<uint16_t, std::shared_ptr<Work>> _sent_commands{};
```

Comment:
Actually, for low numbers a `std::vector<std::pair<uint16_t, Work>>` would likely be faster.

Reply:
I agree that it's more expressive. So, let's leave it as you have it and profile and optimize later if required. We can for instance write an abstraction for ordered and unordered map and implement it with a vector inside.